pkg/plugin/plugin.go

```diff
@@ -122,7 +122,16 @@ func setupVeth(contNetns ns.NetNS, contIfaceName string, requestedMac string, mt
 	// this we will make sure that both ends of the veth pair will be removed
 	// when the container is gone.
 	err := contNetns.Do(func(hostNetns ns.NetNS) error {
-		hostVeth, containerVeth, err := ip.SetupVeth(contIfaceName, mtu, hostNetns)
+		var containerMac net.HardwareAddr
```

Comment:
I think I understand the split now. I was going to suggest we configure the "random" MAC through `SetupVeth` too, but that would not allow us to deal with a conflict... With your change, this function becomes even harder to read (to be fair, it was not great even before you posted this :). 

Could you please move this block and the one where we assign random IP to two isolated functions? It would make this code block more readable and it would also help us decouple the shared variables we have here.

I won't fight for this as your PR seems valid, so feel free to reject my request, but please consider it.

Reply:
Removed the ParseMAC because it is already checked in ```ip.SetupVeth()```.
https://github.com/containernetworking/plugins/blob/2a9114d1af50e23ccb020f585d2c38d7e1fc36f2/pkg/ip/link_linux.go#L46-L50

And removed MAC address assignment because ```ip.SetupVeth()``` assigns MAC addresswhen it is not specified.

Reply:
Great catch!